backend/services/grading_embedding_service.py

```
@@ -0,0 +1,388 @@
+                    'content': chunk,
+                    'embedding': None  # Will be filled
+                })
+                
+                # Create async task for embedding
+                embedding_tasks.append(self._get_embedding_async(chunk, f"material_{material_id}_chunk_{i}"))
+            
+            # Execute all embedding tasks in parallel
+            embeddings = await asyncio.gather(*embedding_tasks)
+            
+            # Combine chunks with embeddings
+            for chunk_dict, embedding in zip(chunk_data, embeddings):
```

Ruff (0.14.1)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)